tsdb/index/tsi1/log_file.go

```diff
@@ -1024,8 +1024,22 @@ type LogEntry struct {
 
 // UnmarshalBinary unmarshals data into e.
 func (e *LogEntry) UnmarshalBinary(data []byte) error {
+	// Wrapper for binary.Uvarint(); returned n is > 0 else err != nil
+	uvarint := func(data []byte) (value uint64, n int, err error) {
+		if len(data) < 1 {
+			err = io.ErrShortBuffer
+		} else if value, n = binary.Uvarint(data); n == 0 || n > len(data) {
+			err = io.ErrShortBuffer
+		} else if n < 0 {
+			err = fmt.Errorf("parsing binary-encoded uint64 value failed; binary.Uvarint() returned %d", n)
+		}
+		return
+	}
```

Comment:
`uvarint` can be promoted to a package-level function since it doesn't (and shouldn't) hold a closure around anything scoped to `(*LogEntry).UnmarshalBinary`.

Reply:
I was going to say that too, but this package is relatively large, so privatizing the function inside `UnmarshalBinary` is at least more readable.

OTOH, `uvarint` is useful anywhere that `binary.Uvarint` is called. There are 10 more uses of `binary.Uvarint` just within the `tsi1` package, and none of them check `n`.

Reply:
PTAL. I left 4 other uses within `tsi1`, they are nested at least two-deep in non-error-returning functions.